### DISCUSSION TITLE
Is it possible to get data returned from root.tsx loader in useRouteData?

### DISCUSSION THREAD

**Discussion by brightpixels** [remix-run/remix#3342] [Q&A] [answered]
I am trying to pass around a "user" object around as global data to any components that needs it from the root.tsx file. Is it possible to get data returned from root.tsx loader in useRouteData?

**Comment by sergiodxa** [comment, marked as answer]
There's no useRouteData hook in Remix, the data returned by the root loader is available inside the root UI in useLoaderData and in any route through useMatches, specifically, the root is always the first match returned by useMatches, so you could do this:

```ts
let [rootMatch] = useMatches()
```

**Reply by brightpixels** [reply]
I posted in the wrong repo! Sorry about that. Question was regarding https://github.com/sergiodxa/remix-utils#useroutedata


Can I reword the problem instead? Let’s say currently in a CRA react app we use Redux to store user preferences like background colour which isn’t persisted to the database and only kept on redux purely for the duration of the visit/session. In the remix world, how should I ‘save’ and retrieve this type of UI data that needs to be accessed by any component or route?

**Reply by sergiodxa** [reply]
Store that in a DB or cookie and read it on the loaders
